File: hw1/cat.java
import java.io.*;

public class cat {
	public static void main(String[] args) throws IOException {
		FileInputStream in = new FileInputStream(FileDescriptor.in);
		FileOutputStream out = new FileOutputStream(FileDescriptor.out);
		int bufSize = 4096;
		byte[] buf = new byte[bufSize];
		int count = 0;
		while ((count = in.read(buf, 0, bufSize)) > -1) {
			out.write(buf, 0, count);
		}
		in.close();
		out.close();
	}
}

File: hw1/cat.c
#include <stdlib.h>
#include <stdio.h>
#include <unistd.h>

int main(void) {
	int buf_size = 4096;
	char buf[buf_size];
	size_t count;
	while ((count = read(0, buf, buf_size)) > 0) {
		write(1, buf, count);
	}
}
